src/test/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterTokenFilterFactoryTests.java

```diff
@@ -170,7 +169,28 @@ public void testPartsAndCatenate() throws IOException {
     String[] expected = new String[]{"Power", "PowerShot", "Shot" };
     Tokenizer tokenizer = new WhitespaceTokenizer();
     tokenizer.setReader(source);
-
     assertTokenStreamContents(filterFactory.create(tokenizer), expected);
   }
+
+  // Test that it doesn't split a 2-chars string
+  public void testTwoChars() throws IOException {
+    Settings indexSettings = Settings.builder()
+            .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
+            .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toString())
+            .put("index.analysis.filter.my_word_delimiter.type", TYPE_NAME)
+            .put("index.analysis.filter.my_word_delimiter.split_on_numerics", true)
+            .put("index.analysis.filter.my_word_delimiter.split_on_case_change", "true")
+            .build();
+    TokenFilterFactory filterFactory = filterFactory(indexSettings, FILTER_NAME);
+
+    String alphanum = "b1";
```

Comment:
I would add a test for greek unicode symbols as well, just to be sure the `isLetterOrDigit` properly detects them.
Otherwise we might have to use the [codepoint version](https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#isLetterOrDigit(int)).

Reply:
Good idea, added